pkg/admissioncontroller/webhook/auth/seed/graph/eventhandler_shoot.go

```diff
@@ -72,6 +73,32 @@ func (g *graph) setupShootWatch(ctx context.Context, informer cache.Informer) er
 	return err
 }
 
+func (g *graph) hasExpectedShootBindingEdges(shoot *gardencorev1beta1.Shoot) bool {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
+
+	shootVertex, found := g.getVertex(VertexTypeShoot, shoot.Namespace, shoot.Name)
+	if !found {
+		return false
+	}
+
+	if shoot.Spec.SecretBindingName != nil {
+		bindingVertex, found := g.getVertex(VertexTypeSecretBinding, shoot.Namespace, *shoot.Spec.SecretBindingName)
+		if !found || !g.graph.HasEdgeFromTo(bindingVertex.id, shootVertex.id) {
+			return false
+		}
```

Comment:
Nit: You could move this into a small function (could also be a var-function within this `hasExpectedShootBindingEdges`) and call it for both `SecretBindingName` and `CredentialsBindingName`.